src/util.tsx

```diff
@@ -8,10 +8,22 @@ import * as timeformat from 'timeformat';
 
 const _ = cockpit.gettext;
 
-export const PodmanInfoContext = React.createContext();
+type PodmanInfoContextType = {
+    cgroupVersion: string,
+    // FIXME: not documented in https://docs.podman.io/en/stable/_static/api.html#tag/system/operation/SystemInfoLibpod
+    registries: unknown,
```

Comment:
Hmm, is this going to get more concrete later? At least `unknown[]` seems to be called for...

Reply:
It's not an array. Indeed we only use one field from it, so I added that now, with adjusting the comment.

Reply:
Ah, indeed, I misread this code in ImageRunModal:

```
Object.keys(this.props.podmanInfo.registries).length !== 0
```

So, isn't it a `Record<string, Registry>`, then?

Reply:
It is a dict indeed, and the code you refered to is indeed nonsense. The length will always be 1.

> src/ImageRunModal.jsx:        const podmanRegistries = registries && registries.search ? registries.search : utils.fallbackRegistries

https://docs.podman.io/en/stable/_static/api.html#tag/system/operation/SystemInfoLibpod at least pins it down as an object, not a list. An actual API call shows:

```
❱❱❱ curl --silent --unix-socket $XDG_RUNTIME_DIR/podman/podman.sock http://d/v5.0.0/libpod/info | jq .registries
{
  "search": [
    "registry.fedoraproject.org",
    "registry.access.redhat.com",
    "docker.io"
  ]
}
```

I'll send a follow-up with fixing that.

Reply:
See #2219